app/models/event_instance.rb

```diff
@@ -64,6 +64,12 @@ def yt_url
     yt_video_id && "https://youtu.be/#{yt_video_id}"
   end
 
+  def channels_for_event event_id
+    return [] if event_id.nil?
+    event = Event.find(event_id)
+    event.slack_channel_codes
+  end
```

Comment:
The event_id parameter is not required as it's an attribute of event_instance. 
I'd suggest updating the method to this using `self` to get the attribute of the instance.
```
  def channels_for_event
    return [] if self.event_id.nil?
    event = Event.find(self.event_id)
    event.slack_channel_codes
  end
```

Then the method can be called from the hangout/youtube notification service method without passing an argument. Example: `@event_instance.channels_for_event`

Reply:
that's great! I was thinking it would be nice to not have to pass it in! I'll update it and push it up :smile_cat:

Reply:
As in option I believe in Ruby you can do `attr_reader :event_id` and then you don't have to use the `self` keyword and can just say `event_id`. Whichever way works for you.

Reply:
You may need to use `attr_reader :event_id` regardless. I'm not 100% sure.